app/controllers/user_controller.php

```diff
@@ -73,4 +74,104 @@ public function logout()
         session_destroy();
         redirect('user/login');
     }
+
+    public function profile()
+    {
+        $user = new User();
+        $user->id = Param::get('user_id');
+        $user->getProfile();
+
+        if(!isset($user->username)) {
+            redirect('notfound/pagenotfound');
+        }
+
+        $threads_followed = array();
+        $thread_followed_id = Follow::getFollowedThreadIds($user->id);
+
+        foreach($thread_followed_id as $thread) {
+            $threads_followed[] = Thread::getById($thread['thread_id']);
+        }
+
+        foreach ($threads_followed as $thread) {
+            $thread->username = User::getUsernameById($thread->user_id);
+        }
+
+        $threads_created = Thread::getByUserId($user->id);
+
+        foreach ($threads_created as $thread) {
+            $thread->username = User::getUsernameById($thread->user_id);
+        }
+
+        $this->set(get_defined_vars());
+    }
+
+    public function edit()
+    {
+        $process = Param::get('process', 'edit');
+
+        switch($process) {
+            case 'account':
+                unset($user);
```

Comment:
Done po https://github.com/renzosunico/MyClassroom/commit/5efeac96e413c70246cede67c14501d3b6146bd9 I also move the $user = new User() outside the switch po. :smile: